src/tests/system/tests/test_authentication.py

```diff
@@ -290,3 +290,74 @@ def test_authentication__user_login_with_overriding_home_directory(
         result = ssh.run("pwd").stdout
         assert result is not None, "Getting path failed!"
         assert result == home_exp, f"Current path {result} is not {home_exp}!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.importance("critical")
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_authentication__user_login_with_modified_PAM_provider_is_offline(
+    client: Client, provider: GenericProvider, method: str, sssd_service_user: str
+):
+    """
+    :title: Authenticate with modified PAM when the provider is offline
+    :setup:
+        1. Create user
+        2. Configure SSSD with "cache_credentials = true" and "krb5_store_password_if_offline = true" and
+            "offline_credentials_expiration = 0"
+        3. Modify PAM configuration files /etc/pam.d/system-auth and /etc/pam.d/password-auth
+        3 Start SSSD
+    :steps:
+        1. Login as user
+        2. Offline, login as user
+        3. Offline, login as user with bad password
+    :expectedresults:
+        1. User can log in
+        2. User can log in
+        3. User cannot log in
+    :customerscenario: False
+    """
+    user = "user1"
+    correct = "Secret123"
+    wrong = "Wrong123"
+    provider.user(user).add(password=correct)
+    if method == "ssh" and "ssh" not in client.sssd.sssd["services"]:
+        client.sssd.sssd["services"] = "nss, pam, ssh"
+    client.sssd.domain["cache_credentials"] = "True"
+    client.sssd.domain["krb5_store_password_if_offline"] = "True"
+    client.sssd.pam["offline_credentials_expiration"] = "0"
+    custom_pam_stack = """
```

Comment:
The test looks good to me. I just do not like multi-line strings like this. Would you consider using `textwrap.dedent` or `inspect.cleandoc` for better readability?

    from inspect import cleandoc
    ...
    def test...
        custom_pam_stack = """
            auth		required	pam_env.so
            auth		sufficient	pam_unix.so try_first_pass likeauth nullok
            auth		required	pam_sss.so forward_pass use_first_pass
            ...
        """
        client.fs.write("/etc/pam.d/system-auth", cleandoc(custom_pam_stack))

What do you think?